File: api/Host.go
package api

import (
	"time"

	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Host 对应 Host.py。
type Host struct {
	*session.ApiGroup
}

// NewHost 创建 Host API 分组。
func NewHost(s *session.Session) *Host {
	return &Host{ApiGroup: session.NewApiGroup(s)}
}

// Info 上报主机信息。对应 host/info (post_set)。
//
// Python 语义：Platform=platform, PlatformVer=userAgent,
// Navigator=version, NavigatorVer=userAgent。
// dateTime 格式化为 "20060102150405"，时区为 "GMT-0700"。
func (h *Host) Info(dateTime time.Time, platform, userAgent, version string) (interface{}, error) {
	return h.S.PostSet("host/info", session.O(
		"Time", dateTime.Format("20060102150405"),
		"Timezone", "GMT"+dateTime.Format("-0700"),
		"Platform", platform,
		"PlatformVer", userAgent,
		"Navigator", version,
		"NavigatorVer", userAgent,
	), false, "api", false, false)
}

File: src/api/host.rs
//! Host API group (`api/Host.py`).

use crate::connection::Connection;
use crate::errors::Result;
use crate::tools::map_of;
use crate::xml::XmlMap;

use super::post_set;

/// Host API group.
pub struct Host<'a> {
    conn: &'a Connection,
}

impl<'a> Host<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        Host { conn }
    }

    /// Send host information to the server.
    ///
    /// The Python method takes a `datetime` and formats `Time` and `Timezone`
    /// from it; here the caller passes those two values pre-formatted.
    ///
    /// * `time` — current time as `%Y%m%d%H%M%S`.
    /// * `timezone` — `GMT` offset string (e.g. `GMT+0200`).
    /// * `platform` — platform information.
    /// * `user_agent` / `version` — client identification strings.
    pub fn info(
        &self,
        time: &str,
        timezone: &str,
        platform: &str,
        user_agent: &str,
        version: &str,
    ) -> Result<String> {
        let body: XmlMap = map_of([
            ("Time", time.to_string()),
            ("Timezone", timezone.to_string()),
            ("Platform", platform.to_string()),
            ("PlatformVer", user_agent.to_string()),
            ("Navigator", version.to_string()),
            ("NavigatorVer", user_agent.to_string()),
        ]);
        post_set(self.conn, "api/host/info", &body)
    }
}
